markdownlint/TOP010_useLazyNumbering/TOP010_useLazyNumbering.js

```diff
@@ -0,0 +1,37 @@
+module.exports = {
+  names: ["TOP010", "ordered-list-for-lazy-numbering"],
+  description: "Ordered lists should use lazy numbering",
+  information: new URL(
+    "https://github.com/TheOdinProject/curriculum/blob/main/markdownlint/docs/TOP010.md"
+  ),
+  tags: ["lazy-numbering"],
+  parser: "markdownit",
+  function: function TOP010(params, onError) {
+    params.parsers.markdownit.tokens.forEach((token) => {
+      if (
+        token.tag === "li" &&
+        token.line !== undefined &&
+        /\d/.test(token.line) &&
+        !/^\s*1\./.test(token.line)
+      ) {
+        const originalText = token.line;
+        // https://regexr.com/80oan to test this regex
+        const updatedText = originalText.replace(/^\s*\d+\./, (match) =>
```

Comment:
Our 3rd regex here `/^\s*\d+\./` could probably saved to a variable outside of the `if` and reused. Then the `if` double regex test condition can be combined into one.